addons/point_of_sale/static/tests/generic_helpers/utils.js

```diff
@@ -25,7 +27,21 @@ export function negateStep(step) {
     };
 }
 export function refresh() {
-    return run(() => window.location.reload(), "refresh page");
+    return run(async () => {
+        await new Promise((resolve) => {
+            const checkTransaction = () => {
+                const activeTransactions = posmodel.data.indexedDB.activeTransactions;
+                if (activeTransactions === 0) {
```

Comment:
```suggestion
                if (activeTransactions <= 0) {
```

take into account the risk of `onerror`, `onabort` and `oncomplete` to be simultaneously happening.